### PR TITLE
When loading on web, match style and show a progress indicator while wasm is loading

### PR DESCRIPTION
**Pull request by jleibs**
### What

Resolves:
 - https://github.com/rerun-io/rerun/issues/2261

According to the new design, here is how the loading state should work:
1. HTML and CSS loads. It renders the skeleton content without any text.
1. WASM loads. It renders the exact same skeleton content without any text. The user doesn’t notice that the rendering method switched from HTML => WASM.
1. If loading takes long (more than 3 seconds) text fades in and provides additional information.
1. When the App UI is ready, the skeleton content fades away (200ms duration) and the app fades in.

The first deployed test of this ended up with 300% progress during the download step, so I've added some extra handling for progress related to compressed streams.At the moment we default to an adhoc compression ratio, but once we land:
 - https://github.com/rerun-io/rerun/issues/2422
 then the progress should be completely accurate again.
 
In order to make all the waiting variants more consistent, this also changes the RRD loading page.

The normal waiting page looks like:
image

Before:
image

After:
image

This PR does not handle progress indicator during the loading of the RRD itself.  The hooks necessary to get access to progress feedback depend on essentially the same work already happening as part of: https://github.com/rerun-io/rerun/pull/2412 and just directly loading the rrd file is generally going to be better than showing a progress indicator. It could be worth revisiting this for things like blueprints with large static data.

This needs to be tested manually on our major OS / browser combinations:

Builds to test:
 - https://app.rerun.io/commit/e15ebb5
 - https://demo.rerun.io/commit/e15ebb5
 
If your internet connection is too fast, use developer tools to throttle connection and enable cache.
image

- Linux:
  - [x] Chrome
  - [x] Firefox
- Mac:
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
- Windows:
  - [x] Chrome
  - [x] Firefox
  - [x] Edge 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)

PR Build Summary: https://build.rerun.io/pr/2421

Docs preview: https://rerun.io/preview/934925b/docs
Examples preview: https://rerun.io/preview/934925b/examples

### CODE REVIEW

**Comment by abey79** [comment]
Mac/Safari: for some reason, nothing is clickable but the rerun menu using the build test link (demo.rerun.io works fine). The loading screen does seem to be fine.

**Comment by nikolausWest** [comment]
Also getting the same behavior that nothing is clickable on mac chrome

**Comment by emilk** [comment]
Screenshot 2023-06-14 at 16 01 54

File: crates/re_viewer/src/ui/wait_screen_ui.rs [hunk 1]
```diff
@@ -30,28 +30,36 @@ pub fn wait_screen_ui(ui: &mut egui::Ui, rx: &Receiver<LogMsg>) {
...
+                        "{}",
+                        paths
+                            .iter()
+                            .format_with(", ", |path, f| f(&format_args!("{}", path.display())))
+                    ),
+                );
             }
             re_smart_channel::SmartChannelSource::RrdHttpStream { url } => {
-                ui.strong(format!("Loading {url}…"));
+                waiting_ui(ui, "Loading...", url);
```
Comment (emilk) [inline crates/re_viewer/src/ui/wait_screen_ui.rs:45]:
```suggestion
                waiting_ui(ui, "Loading…", url);
```

**Comment by jleibs** [comment]
@emilk
> I guess we should clamp the display to at most 99%.

It looks like you're testing an old build. The fixes for that are in there but on commit: https://app.rerun.io/commit/e15ebb5

**Comment by emilk** [comment]
> @emilk
> 
> > I guess we should clamp the display to at most 99%.
> 
> It looks like you're testing an old build. The fixes for that are in there but on commit: https://app.rerun.io/commit/e15ebb5

That makes sense now. I just clicked https://build.rerun.io/pull_request/2421/index.html and assumed newest at the bottom 🤦

File: scripts/demo_assets/static/index.js [hunk 2]
```diff
@@ -17,22 +49,86 @@ function load_wasm() {
...
+    document.getElementById("center_text").classList.add("visible");
+  }, 1500);
+
+  async function wasm_with_progress() {
+    const response = await fetch('./re_viewer_bg.wasm');
+    // Use the uncompressed size
+    var content_length;
+    var content_multiplier = 1;
+    // If the content is gzip encoded, try to get the uncompressed size.
+    if (response.headers.get('content-encoding') == 'gzip') {
```
Comment (emilk) [inline scripts/demo_assets/static/index.js:79]: why just `gzip`, and not `compress`, `deflate`, br`, etc?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding

**Review by emilk — APPROVED** [review]
The code duplication makes me sad, but otherwise LGTM

File: scripts/demo_assets/static/index.js (hunk 2 above)
Comment (jleibs) [reply scripts/demo_assets/static/index.js:79]: gzip is what gcloud happens to do. I'll add support for more encodings in the future:
https://github.com/rerun-io/rerun/issues/2452

**Pull request merged**